vowpalwabbit/main.cc

```diff
@@ -131,6 +131,9 @@ int main(int argc, char* argv[])
 
     for (vw* v : alls)
     {
+      if (v->p->exc_ptr) {
+        std::rethrow_exception(v->p->exc_ptr);
```

Comment:
I'm not sure I follow why the exception needs to be stashed and rethrown? Why can it not just bubble up out of the parser?

Reply:
I was under the impression that this is slightly safer. Don't exceptions that are not caught in `std::thread` just result in the program being `std::terminate`d?

Reply:
Yes they do, but you'll notice there are two catch statements in main which make sure the exception info is printed before exiting. I think letting the exception bubble up is the most idiomatic way of handling this. It also makes library use of the function much simpler